File: frontend/src/tests/data/scenarios.js
export function authResponseScenario() {
  return {
    access_token: 'token-1',
    user: {
      id: 'user-1',
      username: 'demo',
      email: 'demo@example.com',
      display_name: 'Demo',
    },
  }
}

export function authStateScenario() {
  return {
    session: { token: '', user: null },
    payment: { history: [] },
  }
}

export function lineInvoiceScenario() {
  return 'merchant: Cafe\namount: 18.40\nreference: INV-LINE'
}

export function jsonInvoiceScenario() {
  return JSON.stringify({
    invoiceId: 'INV-1',
    merchantName: 'Cafe',
    amount: 12.5,
    currency: 'EUR',
  })
}

export function uriInvoiceScenario() {
  return 'upi://pay?pn=Cafe&am=10.00&cu=eur&tr=ABC-1'
}

File: backend/tests/data/scenarios.py
from __future__ import annotations

import json


def auth_registration_payload(username: str = "demo-user", email: str = "demo@example.com") -> dict[str, str]:
    return {
        "username": username,
        "email": email,
        "password": "demo-pass-123",
        "display_name": "Demo User",
    }


def invoice_payload(reference: str = "INV-TEST") -> dict[str, object]:
    return {
        "invoice": {
            "invoiceId": reference,
            "merchantName": "Cafe Test",
            "description": "Test invoice",
            "currency": "EUR",
            "countryCode": "DE",
            "amount": 12.5,
            "amountMinor": 1250,
            "rawPayload": json.dumps({"reference": reference}),
        }
    }
